polars/polars-sql/src/context.rs

```diff
@@ -281,10 +279,29 @@ impl SQLContext {
         };
 
         // Apply optional 'distinct' clause
-        if select_stmt.distinct {
-            Ok(lf.unique(None, UniqueKeepStrategy::Any))
-        } else {
+        lf = match &select_stmt.distinct {
+            Some(Distinct::Distinct) => lf.unique(None, UniqueKeepStrategy::Any),
+            Some(Distinct::On(exprs)) => {
+                let gb_cols = exprs
+                    .iter()
+                    .map(|e| parse_sql_expr(e, self))
+                    .collect::<PolarsResult<Vec<_>>>()?;
+                // DISTINCT ON applies the ORDER BY before the operation.
+                if !query.order_by.is_empty() {
+                    lf = self.process_order_by(lf, &query.order_by)?;
+                }
+                return Ok(lf.groupby_stable(gb_cols).agg(vec![col("*").first()]));
```

Comment:
I think we can use `lf.unique(UniqueKeepStrategy::First)`

Reply:
@ritchie46 I don't know if it helps, but doing an `order by` followed by `first()` or `last()` is the logical equivalent of doing a `min`/`max`, which means you wouldn't have to maintain order.